### PR TITLE
Support for installing different ubuntu kernel types.

### PR DESCRIPTION
**Pull request by LiliDeng**
I tested proposed-edge and proposed-azure kernel, I need wait for new kernel available in esm and fips to test with kernel url and public key, but you can review code.

### CODE REVIEW

**Review by squirrelsc — DISMISSED** [review]

File: Libraries/CommonFunctions.psm1 [hunk 1]
```diff
@@ -710,10 +710,21 @@ function Install-CustomKernel ($CustomKernel, $allVMData, [switch]$RestartAfterU
 	try {
 		$currentKernelVersion = ""
 		$global:FinalKernelVersion = ""
+		# kerneltype;publickey:keyvalue;kernelurl:http
 		$CustomKernel = $CustomKernel.Trim()
+		$CustomKernelArray = $CustomKernel.split(";")
```
Comment (squirrelsc) [inline Libraries/CommonFunctions.psm1:715]: V3 supports those scenarios already, can you try V3?

**Comment by LiliDeng** [comment]
@lizzha please review, I already tested it.

**Review by lizzha — APPROVED** [review]

File: Libraries/CommonFunctions.psm1 [hunk 2]
```diff
@@ -710,10 +710,21 @@ function Install-CustomKernel ($CustomKernel, $allVMData, [switch]$RestartAfterU
...
+		}
+		if ($CustomKernelArray.length -eq 3) {
+			$CustomKernelKey = $CustomKernelArray[1]
+			$CustomKernelUrl = $CustomKernelArray[2]
+		}
 		# when adding new kernels here, also update script customKernelInstall.sh
 		$SupportedKernels = "ppa", "proposed", "proposed-azure", "proposed-edge",
-			"latest", "linuxnext", "netnext", "upstream-stable"
+		"latest", "linuxnext", "netnext", "upstream-stable", "linux-image-azure-lts-18.04",
+		"esm", "linux-azure-fips", "linux-image-azure-lts-20.04", "linux-image-azure-fde"
```
Comment (lizzha) [inline Libraries/CommonFunctions.psm1:727]: could you add description, which custom kernel options require public key and kernel URL?

File: Libraries/CommonFunctions.psm1 (hunk 1 above)
Comment (LiliDeng) [reply Libraries/CommonFunctions.psm1:715]: sure

File: Libraries/CommonFunctions.psm1 (hunk 2 above)
Comment (LiliDeng) [reply Libraries/CommonFunctions.psm1:727]: sure

**Pull request merged**